src/main/kotlin/calculator/FormulaConverter.kt

```diff
@@ -0,0 +1,83 @@
+package calculator
+
+typealias FormulaConverterValidator = FormulaConverter.Validator
+
+class FormulaConverter {
+    fun convert(formula: String): FormulaElements {
+        formula.shouldValidateFormula()
+
+        val formulas = formula.split(Validator.formulaDelimiter).iterator()
+        return FormulaElements(
+            formulas.next().toDouble(),
+            formulas.asSequence().toList().chunked(2).map {
+                FormulaElement(
+                    type = Operation.of(it[0])!!,
+                    value = it[1].toDouble()
+                )
+            }
+        )
+    }
+
+    private fun String.shouldValidateFormula() = Validator.checkValidateFormulaOrThrows(this)
+
+    object Validator {
+        const val formulaDelimiter = " "
+
+        fun checkValidateFormulaOrThrows(formula: String) {
+            formula.shouldNotEmpty()
+            formula.shouldContainNumber()
+            formula.numbersShouldValidPositions()
+            formula.operationsShouldValidPositions()
+            formula.shouldOddNumberDivided()
+        }
+
+        private fun String.shouldNotEmpty() = require(isNotEmpty())
+
+        private fun String.shouldContainNumber() = require(split().any { it.toDoubleOrNull() != null })
+
+        private fun String.numbersShouldValidPositions() = split().filterIndexed { index, _ -> index.isEven() }
+            .forEach { require(it.toDoubleOrNull() != null) }
+
+        private fun String.operationsShouldValidPositions() = split().filterIndexed { index, _ -> !index.isEven() }
+            .forEach { require(Operation.of(it) != null) }
+
+        private fun String.shouldOddNumberDivided() = require((split().size % 2) == 1)
+
+        private fun String.split() = split(formulaDelimiter)
+
+        private fun Int.isEven() = (this % 2) == 0
+    }
+}
+
+typealias Calculation = (leftOperand: Double, rightOperand: Double) -> Double
+
+enum class Operation(val op: String, val calculation: Calculation) {
```

Comment:
축약어를 사용하시면 다른 개발자가 코드를 볼때 변수의 의미를 파악하기 힘들 수 있어요.
너무 긴 이름이 아니라면 변수를 잘 표현할 수 있는 이름을 지어주는 건 어떨까요?
```suggestion
enum class Operation(val symbol: String, val calculation: Calculation) {
```